src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -2510,20 +2564,69 @@ public void Inheritance_07()
     public abstract int X { get; }
     public virtual int Y { get; }
 }
-data class B(int X, int Y) : A
+abstract data class B1(int X, int Y) : A
+{
+}
+data class B2(int X, int Y) : A
 {
 }";
             var comp = CreateCompilation(source);
             comp.VerifyDiagnostics(
-                // (6,12): error CS0534: 'B' does not implement inherited abstract member 'A.X.get'
-                // data class B(int X, int Y) : A
-                Diagnostic(ErrorCode.ERR_UnimplementedAbstractMethod, "B").WithArguments("B", "A.X.get").WithLocation(6, 12));
+                // (9,12): error CS0534: 'B2' does not implement inherited abstract member 'A.X.get'
+                // data class B2(int X, int Y) : A
+                Diagnostic(ErrorCode.ERR_UnimplementedAbstractMethod, "B2").WithArguments("B2", "A.X.get").WithLocation(9, 12));
+
+            AssertEx.Equal(new string[0], GetProperties(comp, "B1").ToTestDisplayStrings());
+            AssertEx.Equal(new string[0], GetProperties(comp, "B2").ToTestDisplayStrings());
         }
 
         [Fact]
         public void Inheritance_08()
         {
             var source =
+@"abstract class A
+{
+    public abstract int X { get; }
+    public virtual int Y { get; }
+    public virtual int Z { get; }
+}
+abstract class B : A
+{
+    public override abstract int Y { get; }
+}
+data class C(int X, int Y, int Z) : B
+{
+}";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
```

Comment:
>VerifyDiagnostics [](start = 17, length = 17)

These diagnostics look incorrect. According to the latest spec at [https://github.com/dotnet/csharplang/pull/3527](https://github.com/dotnet/csharplang/pull/3527) 'Each "matching" inherited abstract accessor is overridden'

Reply:
[https://github.com/dotnet/roslyn/issues/44785](https://github.com/dotnet/roslyn/issues/44785)

---
In reply to: [434217176](https://github.com/dotnet/roslyn/pull/44705#discussion_r434217176) [](ancestors = 434217176)